src/iterative_sorting/iterative_sorting.py

```diff
@@ -1,28 +1,39 @@
 # TO-DO: Complete the selection_sort() function below 
+array = [11,4,8,5,4,5,2,7,1, 100]   
+
 def selection_sort( arr ):
     # loop through n-1 elements
     for i in range(0, len(arr) - 1):
-        cur_index = i
-        smallest_index = cur_index
+        smallest_index = i
+    
         # TO-DO: find next smallest element
-        # (hint, can do in 3 loc) 
-             
-
-
-
+        # (hint, can do in 3 loc)
         # TO-DO: swap
+        for x in range(i + 1, len( arr )):
+            if arr[smallest_index] > arr[x]:
+                smallest_index = x
 
+        arr[i], arr[smallest_index] =  arr[smallest_index], arr[i]
 
 
 
     return arr
 
+print(selection_sort(array))
+
 
 # TO-DO:  implement the Bubble Sort function below
 def bubble_sort( arr ):
+    for i in range(len(arr) - 1):
```

Comment:
the traditional way to do a bubble sort is to utilize a boolean variable which can act as a flag to know if any sorting has happened in the list. this way, if you are given a mostly sorted list or a completely sorted list, the function doesn't treat it as a unsorted list and iterate through the whole thing only to make no changes.

the idea is to use a while loop. like this:

made_changes = True
while made_changes:
made_changes = False
(for loop)
(if statement/comparison and if swapping happens set made_changes to True)

this is going to be important for the sprint challenge this week.